clock_exercise/clockmanager.h

```diff
@@ -7,48 +7,50 @@ class ClockManager : public QObject
 {
     Q_OBJECT
 
+    // ↓↓↓ properties
     Q_PROPERTY(QString dateText MEMBER m_dateText NOTIFY dateTextChanged)
     Q_PROPERTY(int minutes MEMBER m_minutes NOTIFY minutesChanged)
     Q_PROPERTY(int hours MEMBER m_hours NOTIFY hoursChanged)
-
-    Q_PROPERTY(int timeOfTheDissolveAnimation MEMBER TIME_OF_THE_DISSOLVE_ANIMATION CONSTANT)
-
     Q_PROPERTY(int timerMinutes MEMBER m_timerMinutes NOTIFY timerMinutesChanged)
     Q_PROPERTY(int timerHours MEMBER m_timerHours NOTIFY timerHoursChanged)
     Q_PROPERTY(bool timerIsInPlay MEMBER m_timerIsInPlay NOTIFY timerIsInPlayChanged)
     Q_PROPERTY(float timerCurrentSeconds MEMBER m_timerCurrentSeconds NOTIFY timerCurrentSecondsChanged)
+    Q_PROPERTY(int dayOfDate MEMBER m_dayOfDate NOTIFY dayOfDateChanged WRITE setDayOfDate)
+    Q_PROPERTY(int monthOfDate MEMBER m_monthOfDate NOTIFY monthOfDateChanged WRITE setMonthOfDate)
+    Q_PROPERTY(int yearOfDate MEMBER m_yearOfDate NOTIFY yearOfDateChanged WRITE setYearOfDate)
+    Q_PROPERTY(QString dayOfDateName MEMBER m_dayOfDateName NOTIFY dayOfDateNameChanged WRITE setDayOfDateName)
+    Q_PROPERTY(QString monthOfDateName MEMBER m_monthOfDateName NOTIFY monthOfDateNameChanged WRITE setMonthOfDateName)
     Q_PROPERTY(int minutesInOneHour MEMBER MINUTES_IN_ONE_HOUR CONSTANT)
     Q_PROPERTY(int hoursInADay MEMBER HOURS_IN_A_DAY CONSTANT)
-    Q_PROPERTY(int alarmDay MEMBER m_alarmDay NOTIFY alarmDayChanged WRITE setAlarmDay)
-    Q_PROPERTY(int alarmMonth MEMBER m_alarmMonth NOTIFY alarmMonthChanged WRITE setAlarmMonth)
-    Q_PROPERTY(int alarmYear MEMBER m_alarmYear NOTIFY alarmYearChanged WRITE setAlarmYear)
-    Q_PROPERTY(QString alarmDayName MEMBER m_alarmDayName NOTIFY alarmDayNameChanged WRITE setAlarmDayName)
-    Q_PROPERTY(QString alarmMonthName MEMBER m_alarmMonthName NOTIFY alarmMonthNameChanged WRITE setAlarmMonthName)
+    Q_PROPERTY(int timeOfTheDissolveAnimation MEMBER TIME_OF_THE_DISSOLVE_ANIMATION CONSTANT)
 
     QString m_dateText;
     int m_minutes;
     int m_hours;
     int m_timerMinutes;
     int m_timerHours;
     bool m_timerIsInPlay;
+    float m_timerCurrentSeconds;
+    int m_dayOfDate;
+    int m_monthOfDate;
+    int m_yearOfDate;
+    QString m_dayOfDateName;
+    QString m_monthOfDateName;
+    // ↑↑↑ properties
 
+    // ↓↓↓ not properties
     QTimer m_timerForRefresh;
-    float m_timerCurrentSeconds;
     int m_timerHoursOnStartTimer;
     int m_timerMinutesOnStartTimer;
-    int m_alarmDay;
-    int m_alarmMonth;
-    int m_alarmYear;
-    QString m_alarmDayName;
-    QString m_alarmMonthName;
+    // ↑↑↑ not properties
```

Comment:
Questi commenti qui in particolare, vicini alle variabili membro, sono i classici commenti che rischiano di desincronizzarsi in tempo zero: se domani non serve più esporre una property quello che faccio è semplicemente rimuovere una Q_PROPERTY dall'alto (supponiamo si tratti della `monthOfDateName`.
Il risultato è che adesso  nella sezione delle property ho ancora una variabile membro che non è più esposta al QML.

In generale aggiungere un commento ha sia dei benefici, che dei costi.
I benefici sono quelli per cui, se ben fatti, rendono il codice più chiaro.
I costi riguardano invece la manutenzione: se non si fa manutenzione a un commento e questo va fuori sync col codice, non sono non hai più il beneficio, adesso hai un malus aggiuntivo dato dal fatto che il commento è ingannevole.
La regola d'oro è che se i benefici apportati da un commento sono marginali, dobbiamo analizzare il bonus a fronte dei costi di manutenzione. Se a priori sappiamo che non vogliamo manutenerli, allora il problema non si pone.

In questo caso, penso che i benefici siano molto piccoli, e le probabilità di andare out of sync molto alte.